velox/docs/functions/presto/conversion.rst

```diff
@@ -919,6 +1032,101 @@ Invalid example
   SELECT cast('-3E+2.1' as decimal(12, 2)); -- Value is not a number
   SELECT cast('3E+' as decimal(12, 2)); -- Value is not a number
 
+Cast to IPADDRESS
+-----------------
+
+From VARCHAR
+^^^^^^^^^^^^
+
+To cast a varchar to IPAddress input string must be in the form of either
+IPV4 or IPV6.
+
+For IPV4 it must be in the form of:
+x.x.x.x where each x is an integer value between 0-255.
+
+For IPV6 it must follow any of the forms defined in `RFC 4291#section-2.2 <https://datatracker.ietf.org/doc/html/rfc4291.html#section-2.2>`_.
+
+Full form:
+
+::
+
+   2001:0DB8:0000:0000:0008:0800:200C:417A
+   2001:DB8:0:0:8:800:200C:417A
+
+Compressed form:
+::
+  2001:DB8::8:800:200C:417A
+
+Alternate form:
+::
+  0:0:0:0:0:0:13.1.68.3
```

Comment:
Seeing the following behavior in Presto:

```
presto:di> select cast('0:0:0:0:0:0:13.1.68.3' as ipaddress);
   _col0
------------
 ::d01:4403
```

Would you confirm that Velox impl matches?

Reply:
I don't think that result is correct on the presto side. It should be `::13.1.68.3`
That is a good case though. I will add a test and update the docs. Might need to open a presto issue on that.

Reply:
At first I thought it should be `13.1.68.3` but actually it should be `::13.1.68.3` as its an IPV6 address not an IPV4 converted IPV6 address.